loopy/transform/iname.py

```diff
@@ -1737,6 +1736,49 @@ def add_inames_to_insn(kernel, inames, insn_match):
 # }}}
 
 
+# {{{ remove_inames_from_insn
+
+@for_each_kernel
+def remove_inames_from_insn(kernel, inames, insn_match):
+    """
+    :arg inames: a frozenset of inames that will be added to the
+        instructions matched by *insn_match*, or a comma-separated
+        string that parses to such a tuple.
+    :arg insn_match: An instruction match as understood by
+        :func:`loopy.match.parse_match`.
+
+    :returns: a :class:`LoopKernel` with the *inames* removed from
+        the instructions matched by *insn_match*.
+
+    .. versionadded:: 2023.0
+    """
+
+    if isinstance(inames, str):
+        inames = frozenset(s.strip() for s in inames.split(","))
```

Comment:
Eh, I think @kaushikcfd is right when he tries to get us away from this comma-separated string stuff.

Reply:
Done in eea9df173e675c7caee7d8f539123293088fbe9a